lib/msf/ui/console/driver.rb

```diff
@@ -706,77 +696,44 @@ def handle_session_tlv_logging(val)
   # Require the appropriate readline library based on the user's preference.
   #
   # @return [void]
-  def choose_readline(opts)
-    # Choose a readline library before calling the parent
-    @rl_err = nil
-    if opts['RealReadline']
-      # Remove the gem version from load path to be sure we're getting the
-      # stdlib readline.
-      gem_dir = Gem::Specification.find_all_by_name('rb-readline').first.gem_dir
-      rb_readline_path = File.join(gem_dir, "lib")
-      index = $LOAD_PATH.index(rb_readline_path)
-      # Bundler guarantees that the gem will be there, so it should be safe to
-      # assume we found it in the load path, but check to be on the safe side.
-      if index
-        $LOAD_PATH.delete_at(index)
-      end
-    end
+  def setup_readline
+    # Only Windows requires a monkey-patched RbReadline
+    return unless Rex::Compat.is_windows
+
+    if defined?(::RbReadline) && !defined?(RbReadline.refresh_console_handle)
+      ::RbReadline.instance_eval do
+        class << self
+          alias_method :old_rl_move_cursor_relative, :_rl_move_cursor_relative
+          alias_method :old_rl_get_screen_size, :_rl_get_screen_size
+          alias_method :old_space_to_eol, :space_to_eol
+          alias_method :old_insert_some_chars, :insert_some_chars
+        end
 
-    begin
-      require 'readline'
```

Comment:
I assume we still want the `require 'readline'` line around? 👀

Reply:
Added back in. `defined?(Readline)` and `defined?(RbReadline)` were both `nil`. 👍